### ISSUE TITLE
Logged exceptions shouldn't contain colour

### ISSUE THREAD

**Issue by majutsushi** [Qix-/better-exceptions#87]
Currently if the output is to a terminal, then an exception that gets logged will also include the colour escape codes. This is not really ideal since they will just show up as raw escape codes in most editors, making the log harder to read. So it would be great if the logged exceptions could skip the colour-coding even if the current terminal supports colours. The output to the terminal itself should still be coloured of course. Thanks!

**Comment by Qix-** [comment]
I'm confused.

> Currently if the output is to a terminal, then an exception that gets logged will also include the colour escape codes.

Yes, correct. That's how terminal colors work :p

> This is not really ideal since they will just show up as raw escape codes in most editors, making the log harder to read.

Which editors? Where?

> So it would be great if the logged exceptions could skip the colour-coding even if the current terminal supports colours.

Why would that be great? That just means no color.

> The output to the terminal itself should still be coloured of course

I'm completely lost. What issue are you facing?

**Comment by majutsushi** [comment]
I mean that exception colours will also be present in log files that get written when using the `logging` module, for example when using `logging.exception("oops")`. These escape codes don't really belong in log files, only in direct terminal output, because editors that you would use to inspect log files are (generally) not terminals.

**Comment by Qix-** [comment]
Ohhh, you mean with the logging module. Okay, I agree.

**Comment by majutsushi** [comment]
Sorry for being unclear earlier!

**Comment by Qix-** [comment]
No problem :)

**Comment by jw-cpnet** [comment]
Yes, I'm facing the same issue. Could you please consider adding an option to disable coloring?

**Comment by lokesh1729** [comment]
@Qix- @majutsushi any update on this??? as i understand we don't want the color ASCII characters to include while logging to files...

**Comment by Qix-** [comment]
@lokesh1729 As is with most OSS, if it is not updated on this issue, then there is no update. We don't secretly discuss/code outside of Github. Please don't spam my issues with comments asking for updates.

**Comment by lokesh1729** [comment]
@Qix- i understand but my intention is not to spam but to contribute... i want to work on this issue, so "update" i mean is wanted to know if someone has already made a progress on it...

**Comment by Qix-** [comment]
Nope, nobody has made progress on it.

**Comment by philippludwig** [comment]
I can’t reproduce this. Whenever the `logging` module is involved, nothing of the output is colored. Maybe someone can provide a MCVE.

**Comment by isaac-philip** [comment]
Hello I c this issue still open? I can have a look

**Comment by PacoMatias89** [comment]
Hi! I’d like to work on this issue.

I can prepare a fix so terminal output keeps colors, but logged exceptions written to files do not include ANSI color codes.

I noticed there is an old submitted PR linked to this issue (#99). Is this still open for a fresh contribution, or would you prefer updating/replacing that approach?

**Comment by shriram-svg** [comment]
Opened #151 with a fresh fix for the shared formatter/file handler ANSI leak. It keeps stderr logging colored while file-handler output stays uncolored.

**Comment by Question86** [comment]
Opened PR #156 for this issue: https://github.com/Qix-/better-exceptions/pull/156\n\nScope:\n- keeps stderr tracebacks colored\n- keeps file handler tracebacks uncolored\n- covers shared-formatter + handler-order regression with a dedicated test (	est/test_file_logging.py).
